packages/vue-app/template/App.js

```diff
@@ -100,6 +101,28 @@ export default {
         }
       }
     },
+    async refreshPageData(showProgressBar) {
+      const context = this.$options.context
+      const pages = getMatchedComponentsInstances(this.$route)
+
+      if (!pages.length) {
+        return
+      }
+      this.$loading && this.$loading.start()
+      const promises = pages.map(async (page) => {
+        if (page.$options.fetch) {
+          await page.$options.fetch(context)
```

Comment:
I think during page load we call fetch and asyncData in parallel. Here we can do the same by pushing promises to an array and await on all